includes/abstracts/llms-abstract-user-engagement.php

```diff
@@ -97,6 +97,65 @@ public function get_related_post_id() {
 		return isset( $meta->post_id ) ? absint( $meta->post_id ) : $this->get( 'related' );
 	}
 
+	/**
+	 * Retrieves the LLMS_Abstract_User_Engagement instance for a given post.
+	 *
+	 * Based on {@see llms_get_certificate()}.
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_Post|int|null $post             A WP_Post object or a WP_Post ID. A falsy value will use the current
+	 *                                           global `$post` object (if one exists).
+	 * @param bool             $preview_template If `true`, allows loading for previewing the template.
+	 * @return LLMS_Abstract_User_Engagement|bool
+	 */
+	protected function get_template_object( $post, $preview_template ) {
+
+		$post = get_post( $post );
+		if ( ! $post ) {
+			return false;
+		}
+
+		if (
+			"llms_my_$this->model_post_type" === $post->post_type ||
+			( "llms_$this->model_post_type" === $post->post_type && $preview_template )
+		) {
+			return new static( $post );
+		}
+
+		return false;
+
+	}
+
+	/**
+	 * Retrieves the earned engagement's template version.
+	 *
+	 * Since LifterLMS 6.0.0, earned engagements are created using the block editor.
+	 *
+	 * Earned engagements created in the classic editor will use template version 1 while any earned engagements
+	 * created in the block editor use template version 2. Therefore an earned engagement that has content
+	 * and no blocks will use template version 1 and any empty earned engagements or those containing blocks
+	 * will use template version 2.
+	 *
+	 * @since [version]
+	 *
+	 * @return integer
+	 */
+	public function get_template_version() {
+
+		$version = empty( $this->get( 'content', true ) ) || has_blocks( $this->get( 'id' ) ) ? 2 : 1;
+
+		/**
+		 * Filters an earned engagement's template version.
+		 *
+		 * @since [version]
+		 *
+		 * @param int $version The template version.
+		 */
+		return apply_filters( "llms_{$this->model_post_type}_template_version", $version, $this );
+
+	}
```

Comment:
I don't believe this belongs here since achievements cannot be edited in the block editor so this applied only to certificates

Reply:
`get_template_version()` is called by `sync()`, which was moved from the `LLMS_User_Certificate` class to `LLMS_Abstract_User_Engagement`. Instead of using this "generic" version, would you prefer to have a version in `LLMS_Abstract_User_Engagement` that always returns 1 and have an overriding version in `LLMS_User_Certificate`?